docs/resources/synthetic_monitoring_check_alerts.md

```diff
@@ -0,0 +1,65 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "grafana_synthetic_monitoring_check_alerts Resource - terraform-provider-grafana"
+subcategory: "Synthetic Monitoring"
+description: |-
+  Manages alerts for a check in Grafana Synthetic Monitoring.
+  These alerts will trigger when your check fails from the specified probes.
+---
+
+# grafana_synthetic_monitoring_check_alerts
+
+Manages alerts for a check in Grafana Synthetic Monitoring.
+
+* [Official documentation](https://grafana.com/docs/grafana-cloud/testing/synthetic-monitoring/configure-alerts/configure-per-check-alerts/)
+* [API documentation](https://github.com/grafana/synthetic-monitoring-api-go-client/blob/main/docs/API.md#alerts)
+
+## Example Usage
+
+```hcl
+resource "grafana_synthetic_monitoring_check" "main" {
+  job     = "Check Alert Test"
+  target  = "https://grafana.com"
+  enabled = true
+  probes  = [1]
+  labels  = {}
+  settings {
+    http {
+      ip_version = "V4"
+      method     = "GET"
+    }
+  }
+}
+
+resource "grafana_synthetic_monitoring_check_alerts" "main" {
+  check_id = grafana_synthetic_monitoring_check.main.id
+  alerts = [{
+    name      = "ProbeFailedExecutionsTooHigh"
+    threshold = 0.5
+    period    = "5m"
+  },
+  {
+    name      = "TLSTargetCertificateCloseToExpiring"
+    threshold = 7
+    period    = ""
+  }]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `check_id` - (Required) The ID of the check to manage alerts for.
+* `alerts` - (Required) List of alerts for the check. Each alert has the following fields:
+  * `name` - (Required) Name of the alert. Must be one of: `ProbeFailedExecutionsTooHigh`, `TLSTargetCertificateCloseToExpiring`.
+  * `threshold` - (Required) Threshold value for the alert.
+  * `period` - (Required) Period for the alert threshold. Required and must be one of: `1m`, `2m`, `5m`, `10m`, `15m`, `20m`, `30m`, `1h` for `ProbeFailedExecutionsTooHigh` alerts. For other alerts, set to `""` (empty string).
```

Comment:
I think the _period_ is more related with the alert type, more than "with the threshold". Also, the min period is by now `5m`. So should this be:
```suggestion
  * `period` - (Required) Period for the alert. Required and must be one of: `5m`, `10m`, `15m`, `20m`, `30m`, `1h` for `ProbeFailedExecutionsTooHigh` alerts. For other alerts, set to `""` (empty string).
```
?

I'm also wondering if we should mention specific alert names or considerations for these, as that means that we'll have to update these docs everytime we add a new alert, or we can just rely on the link to the Grafana Cloud docs and expect to provide the full documentation there?

Reply:
Good call. Removed alert names and adjusted the min period.